processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -216,18 +217,12 @@ public File mergeQueryableIndex(
       ProgressIndicator progress
   ) throws IOException
   {
+    List<IndexableAdapter> indexAdapteres = new ArrayList<IndexableAdapter>();
+    for (QueryableIndex index : indexes) {
```

Comment:
Yes, during the merge, there is a iteration like 
for (dimValue : allDimValues) {
   for (int j = 0; j < indexes.size(); ++j) {
   }
}
The function will get applied "cardinality" times, and there is some disk IO in the constructor of QueryableIndexIndexableAdapter.